src/lib/network_pool/snark_pool.ml

```diff
@@ -1133,34 +1133,10 @@ let%test_module "random set test" =
             Mock_snark_pool.For_tests.get_rebroadcastable resource_pool
               ~has_timed_out:(Fn.const `Ok)
           in
-          let () =
-            let open Mock_snark_pool.Resource_pool.Diff in
-            let compare x y =
-              match (x, y) with
-              | Add_solved_work (stmt1, _), Add_solved_work (stmt2, _) ->
-                Transaction_snark_work.Statement.compare stmt1 stmt2
-              | _ ->
-                assert false
-            in
-            [%test_eq: Mock_snark_pool.Resource_pool.Diff.t list]
-              (List.sort  ~compare rebroadcastable3)
```

Comment:
Didn't you add these to make the unit test pass?

Reply:
I added the sorting, yes. The equality tests were deleted in `compatible`, for whatever reason.

Reply:
I think the equality checks have been moved to `check_work`.